build.yaml

```diff
@@ -3665,7 +3679,7 @@ configs:
     DEFINES: NDEBUG
   dbg:
     CPPFLAGS: -O0
-    DEFINES: _DEBUG DEBUG
+    DEFINES: _DEBUG DEBUG PO_DEBUG
   easan:
```

Comment:
PO_DEBUG  'polling object debug'. Mainly to make sure we are correctly using the poll_obj structures. 

In the following structure `obj_type` is redundant in regular builds - but having it will help us add some extra checks to make sure we are passing the correct polling object types to the functions (like `add_poll_obj`).

typedef struct poll_obj {
#ifdef PO_DEBUG
   poll_obj_type obj_type; 
#endif
    gpr_mu mu;
    struct polling_island *pi;
 } poll_obj;

If it were just debug statements, I would have been find using DEBUG flag. Since i am modifying the structure here by adding a new field, it didn't feel right to use DEBUG flag i.e i want to poll_object structure to be the same in opt and dbg builds.